### PR TITLE
Genaral website update/overhaul

### LINKED ISSUE #12: Improve general website security and privacy
See full results from Webbkoll: https://webbkoll.dataskydd.net/en/results?url=http%3A%2F%2Fopenandroidinstaller.org

Most important changes to be made:
- [x] Add CSP
  - Defines how, what and from where stuff on a website is loaded (like CSS, images, JS, etc.)
- [x] Add Referrer Policy
  - Adjusts the referrer header which is send to the next web server when clicking a external link

Using `.htaccess`

### LINKED ISSUE #13: Cleaner coding suggestions
The HTML validator suggests multiple corrections. I checked every page on its own:
- [Homepage](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org)
- [Download page](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org%2Fdownload.html)
- [FAQ](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org%2Ffaq.html)
- [Feedback](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org%2Ffeedback.html)
- [Privacy](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org%2Fprivacy.html)
- [Supported devices](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org%2Fsupported-devices.html)
- [Imprint](https://validator.w3.org/nu/?showsource=yes&showoutline=yes&showimagereport=yes&doc=https%3A%2F%2Fopenandroidinstaller.org%2Fimprint.html)

Please don't take this as "Do everything it suggests", but rather cherry picking the corrections to make the OAI Website better.

### PR DESCRIPTION
**Pull request by MagicLike**
- [x] Update links
  - [x] Add `rel="noopener"` to every link
  - [x] Update outdated links
  - [x] Remove trailing slashes
  - [x] Remove dead links
- [x] Spelling
- [x] Code formatting
- [x] Update FAQ
- [x] Remove inline CSS
- [x] Comply with HTML standards
- [x] Add donation buttons/widgets
- [x] Rearrange devices according to README
- [x] Adding border-radius to menu to fit website theme (#18)
- [x] Use transition: ease-in on hover for certain areas like the menu, the Download buttons, etc. (#18)
- [x] #13
- [x] #12
- [x] Document website build workflow in README
  - [x] Create a GitHub Actions build workflow (see in action [here](https://github.com/Magic-Tests/openandroidinstaller-dev.github.io/actions))
- [x] Add 404 page

### CODE REVIEW

File: templates/download.html [hunk 1]
```diff
@@ -114,7 +115,7 @@ <h2 class="has-text-white mt-2 mb-3 is-size-1 is-size-3-mobile has-text-weight-b
       <div class="column is-5 ml-auto">
         <div class="mx-auto box p-6 has-background-light has-text-centered">
           <a class="button is-link is-fullwidth" id="contribute-button"
-            href="mailto:help@openandroidinstaller.org">Write a mail</a>
+            href="mailto:help@openandroidinstaller.org">Write a email</a>
```
Comment (tsterbak) [inline templates/download.html]:
```suggestion
            href="mailto:help@openandroidinstaller.org">Write an Email</a>
```

Comment (MagicLike) [reply templates/download.html]: fixed in 431990f

**Comment by MagicLike** [comment]
The only thing missing is https://github.com/openandroidinstaller-dev/openandroidinstaller-dev.github.io/issues/16#issuecomment-1729855472

@tsterbak

**Comment by tsterbak** [comment]
Are you sure the new workflow works?

Also, it would be great if the build and deployment would be triggered by adding commits to the main branch, and not only through PRs.

File: templates/privacy.html [hunk 2]
```diff
@@ -160,8 +161,8 @@ <h2 class="has-text-white mt-2 mb-3 is-size-1 is-size-3-mobile has-text-weight-b
       </div>
       <div class="column is-5 ml-auto">
         <div class="mx-auto box p-6 has-background-light has-text-centered">
-          <a class="button is-link is-fullwidth" id="contribute-button"
-            href="mailto:help@openandroidinstaller.org">Write a mail</a>
+          <a class="button is-link is-fullwidth hover-transition" id="contribute-button"
+            href="mailto:hello@openandroidinstaller.org">Write an e-mail</a>
```
Comment (tsterbak) [inline templates/privacy.html]: There is a reason why this is `help@`. I want to know where people came from :)

**Comment by MagicLike** [comment]
> Also, it would be great if the build and deployment would be triggered by adding commits to the main branch, and not only through PRs.

fixed!

File: templates/privacy.html (hunk 2 above)
Comment (MagicLike) [reply templates/privacy.html]: Ooops O.o

Comment (MagicLike) [reply templates/privacy.html]: fixed now

**Comment by MagicLike** [comment]
> Are you sure the new workflow works?

I tested it multiple times, it should work...

File: templates/privacy.html (hunk 2 above)
Comment (tsterbak) [reply templates/privacy.html]: No worries, It was written nowhere :) So you couldn't know

**Review by tsterbak — APPROVED** [review]
LGTM :)

**Comment by tsterbak** [comment]
I'm planning to push the next release of the Installer on Friday (next week) and then publish this PR.

Thanks a lot for cleaning and improving the website! :blush:

**Comment by MagicLike** [comment]
> I'm planning to push the next release of the Installer on Friday (next week) and then publish this PR.
> 
> Thanks a lot for cleaning and improving the website! 😊

No problem! :D - But there is still stuff to do after that ^^

**Comment by tsterbak** [comment]
Did you test the new github workflow? Because every time I touch this stuff something breaks if I don't test it.
Otherwise, I will remove the workflow update from this PR and try to get it in after the release.

**Comment by MagicLike** [comment]
> Did you test the new github workflow? Because every time I touch this stuff something breaks if I don't test it. Otherwise, I will remove the workflow update from this PR and try to get it in after the release.

I did - multiple times, One example can be seen here: https://github.com/Magic-Tests/openandroidinstaller-dev.github.io/actions

But if you feel uncomfortable about it, feel free to not include it for now!

**Comment by tsterbak** [comment]
> I did - multiple times, One example can be seen here: https://github.com/Magic-Tests/openandroidinstaller-dev.github.io/actions

Okay great, great! 😊 looks perfect 👏 thank you for clarifying.

> But if you feel uncomfortable about it, feel free to not include it for now!

No, I want it in and it looks solid 🙂

**Comment by tsterbak** [comment]
The build workflow didn't work properly, since it has no permission to push to main. I guess this is a general issue and cannot be avoided if the main branch is protected and only allows commits through PRs... Maybe pushing to the deployment branch might work? I'm not sure

**Comment by MagicLike** [comment]
> The build workflow didn't work properly, since it has no permission to push to main. I guess this is a general issue and cannot be avoided if the main branch is protected and only allows commits through PRs... Maybe pushing to the deployment branch might work? I'm not sure

I'm gonna do some research later 🤔

**Comment by tsterbak** [comment]
> I'm gonna do some research later 🤔

Seems to be an annoying issue :D

But other than that it would have worked great! And it also looks great now, thanks a lot :)

**Comment by MagicLike** [comment]
> > I'm gonna do some research later 🤔
> 
> Seems to be an annoying issue :D
> 
> But other than that it would have worked great! And it also looks great now, thanks a lot :)

Hmmm it should be possible for you to allow the Actions user to commit directly to main in the branch protection section :thinking: (Should be `github-actions[bot]`)

**Comment by tsterbak** [comment]
> Hmmm it should be possible for you to allow the Actions user to commit directly to main in the branch protection section 🤔 (Should be github-actions[bot])

Where did you get that info from? My understanding was, that has been removed, since it technically allows to bypass branch protection for anyone who can create branches on the repo.

**Comment by MagicLike** [comment]
> > Hmmm it should be possible for you to allow the Actions user to commit directly to main in the branch protection section 🤔 (Should be github-actions[bot])
> 
> Where did you get that info from? My understanding was, that has been removed, since it technically allows to bypass branch protection for anyone who can create branches on the repo.

I made a quick search and found this: https://overflow.adminforge.de/questions/69263843/how-to-push-to-protected-main-branches-in-a-github-action#72515781

**Comment by tsterbak** [comment]
> I made a quick search and found this: https://overflow.adminforge.de/questions/69263843/how-to-push-to-protected-main-branches-in-a-github-action#72515781

Going forward, this will allow anyone allowed to create branches on the repo to basically commit with admin privileges. And some comments say it doesn't even work.

I guess I rather prefer to just have html commited in the public folder on main deployed? Seems very easy and I build the files locally first either way. :thinking:  I will think about it

**Comment by MagicLike** [comment]
> > I made a quick search and found this: https://overflow.adminforge.de/questions/69263843/how-to-push-to-protected-main-branches-in-a-github-action#72515781
> 
> Going forward, this will allow anyone allowed to create branches on the repo to basically commit with admin privileges. And some comments say it doesn't even work.
> 
> I guess I rather prefer to just have html commited in the public folder on main deployed? Seems very easy and I build the files locally first either way. 🤔 I will think about it

wait, there should also be an option to let GH Actions run alongside a PR, for example, when it gets approved it will build the website on the PR branch and then it can get merged...

**Comment by tsterbak** [comment]
> wait, there should also be an option to let GH Actions run alongside a PR, for example, when it gets approved it will build the website on the PR branch and then it can get merged...

That would be great I guess :)
Could we even get a preview deployment of the website in this way?

**Comment by MagicLike** [comment]
> > wait, there should also be an option to let GH Actions run alongside a PR, for example, when it gets approved it will build the website on the PR branch and then it can get merged...
> 
> That would be great I guess :)
> Could we even get a preview deployment of the website in this way?

That should also be possible... - But I will need to investigate that further in the next few days (I have never done that before)

**Pull request merged**